src/main/java/com/nique1/musicmanagementsystem/song/SongController.java

```diff
@@ -0,0 +1,32 @@
+package com.nique1.musicmanagementsystem.song;
+
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.server.ResponseStatusException;
+
+import java.util.List;
+
+@RestController
+@RequestMapping("/songs")
+public class SongController {
+    private final SongService songService;
+
+    public SongController(SongService songService) {
+        this.songService = songService;
+    }
+
+    @GetMapping("{id}")
+    public SongEntity findSongBySongId(@PathVariable int id) {
+        return songService.getSongsById(id)
+                .orElseThrow(() -> new ResponseStatusException(HttpStatus.NOT_FOUND));
+    }
+
+    @GetMapping("search")
+    public List<SongEntity> findSongByArtistTrackOrYear(@RequestParam(required = false) String artistName,
```

Comment:
Zalecana forma nazewnictwa query parametrów to raczej kebab-case, np. `artist-name`